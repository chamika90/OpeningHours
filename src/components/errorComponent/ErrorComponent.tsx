import React from 'react';
import {StyleSheet, Text} from 'react-native';
import {theme} from '../../config/theme';

const {colors} = theme;

type ErrorComponentProps = {
  message: string;
};

const ErrorComponent = ({message}: ErrorComponentProps) => {
  return <Text style={styles.errorMessage}>{message} </Text>;
};

export default ErrorComponent;
const styles = StyleSheet.create({
  errorMessage: {
    color: colors.primaryTextColor,
  },
});
