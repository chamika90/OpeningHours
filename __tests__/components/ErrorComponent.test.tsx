import 'react-native';
import React from 'react';
import {render} from '@testing-library/react-native';
require('jest-fetch-mock').enableMocks();
import ErrorComponent from '../../src/components/ErrorComponent/ErrorComponent';

describe('Opening Hours ', () => {
  it('should match the opening hours snapshot', () => {
    const {toJSON} = render(
      <ErrorComponent message={'Something went wrong'} />,
    );
    expect(toJSON()).toMatchSnapshot();
  });

  it('should render the title ', () => {
    const {getByTestId} = render(
      <ErrorComponent message={'Something went wrong'} />,
    );
    expect(getByTestId('error')).toBeTruthy();
  });
});
