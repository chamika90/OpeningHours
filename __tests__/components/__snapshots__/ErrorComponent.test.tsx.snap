// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Opening Hours  should match the opening hours snapshot 1`] = `
<Text
  style={
    Object {
      "color": "#202125",
    }
  }
  testID="error"
>
  Something went wrong
</Text>
`;
